auslib/db.py

```diff
@@ -397,14 +397,15 @@ def _prepareInsert(self, trans, changed_by, **columns):
         if self.versioned:
             data['data_version'] = 1
         query = self._insertStatement(**data)
-
-        if self.onInsert:
-            self.onInsert(self, "INSERT", changed_by, query, trans)
-
         ret = trans.execute(query)
         if self.history:
             for q in self.history.forInsert(ret.inserted_primary_key, data, changed_by):
                 trans.execute(q)
+        if self.onInsert:
```

Comment:
Moving this code means that we're not sending e-mail until after the row has been modified. This is a really nice side effect of this patch. If you have time, do you think you can follow-up and do the same for update and delete?